### DISCUSSION TITLE
Discrepencies with NumValenceElectrons

### DISCUSSION THREAD

**Discussion by jennijones** [rdkit/rdkit#7111] [Q&A]
I am running NumValenceElectrons() on the same molecule; in one case with the RDKit mol object direct from UFF optimisation, and in the other case converting that mol to xyz format, then back to mol, then adding connectivity with DetermineConnectivity(). 

I am getting 168 in the first case and 167 in the second case. Making the same calculation via DFT, the molecule does have 167 electrons. Why might the number returned in the first case be incorrect?

**Comment by greglandrum** [comment]
@jennijones please provide the specific example that you're using, ideally with code, so that we can try and figure out what's going on.

**Reply by jennijones** [reply]
Hi @greglandrum, sorry, my code is below. I'm not sure how to upload the original mol object which is used in the first case but this is the xyz file for the mol I'm having the issue with, having converted mol to xyz using `Chem.MolToXYZFile`:  [DABNA-1_4562.xyz.zip](https://github.com/rdkit/rdkit/files/14181865/DABNA-1_4562.xyz.zip)

The code in the first case is:
`print(NumValenceElectrons(mol))`
(returns 168)

And in the second case:
```
mol = MolFromXYZFile('FF_geopt_xyz_files/DABNA-1_4562.xyz')
conn_mol = Chem.Mol(mol)
rdDetermineBonds.DetermineConnectivity(conn_mol)
print(NumValenceElectrons(conn_mol))
```
(returns 167)
